design-guidelines/form-ux.md

```diff
@@ -0,0 +1,189 @@
+
+Render a form on it's own page if the form:
+
+- is complex and has many fields
+- will be used to complete an action that is very important
+
+### Section of a page
+
+A form may be rendered on a page alongside additional content to avoid taking users out of their current flow.
+
+Render a form alongside other content if:
+
+- the user may need to reference page content to help them fill out the form correctly
+- the form is not so complex that it deserves it's own page
+- it would be too disruptive to take a user away from the current page
+- the page has content that is related to the form
+
+### Dialog modal
+
+A form may be rendered in a dialog modal to keep the user on their current page while directing their focus to a form. The first input of the form should receive focus when the modal is opened.
+
+Render a form in a dialog modal if:
+
+- there are very few fields in the form
+- there is no logical place to render the form on the current page
+- the user won't need to reference content that will be covered by the dialog
+
+### Inline inputs
+
+In lieu of rendering an entire form, a single input can be used in place of the data it controls. Inputs would replace the data once the user has explicity toggled edit mode.
+
+Use an inline input if it would be too disruptive to show an entire form or take the user to a new page.
+
+## Structure and hierarchy
+
+Forms are designed with a consistent hierarchy and structure to make it easier for users to scan the inputs and submit their data.
+
+Every form will have at least one section, but form sections should always be used when a complex form can be broken down into distinct sections of related fields.
+
+For more information about the parts that make up a form, see the Storybook documentation for each component under the "Form structure" section.
+
+### Field order
+
+Order fields by their relative level of importance. The inputs that are most critical for the user to complete their task should come first.
+
+### Field grouping
+
+Fields that are closely related should be visually grouped together.
+
+For more information on grouped fields, see the Storybook documentation for the `FieldList` and `FieldGroup` components.
+
+### Form actions
+
+A form should contain at least two buttons: one to submit the form, and one to navigate away from the form without submitting any of the input.
+
+These buttons should be highly visible, and easy for the user to press.
+
+### Multi-part forms
+
+Forms should be kept as succinct as possible, but some forms require a lot of data input. Break complex forms into multiple forms when it will help the user focus on what data they're inputting.
+
+#### Tabbed form pattern
+
+Break a form into different tabs when a form is used to configure an object with so many data points that there is no way to keep the fields on one screen without overwhelming the user. For example: a form that is used to deploy an addon with many configuration options.
```

Comment:
> Break a form into different tabs when a form is used to configure an object with so many data points that there is no way to keep the fields on one screen without overwhelming the user

just a style thing: i sometimes feel like i lost brain cycles when reading a long sentence that does not have significant information for some words. maybe we can shorten the first sentence to 

"Break a form into different tabs when its fields don't fit on one screen without overwhelming the user."

i know that it does not even cut the sentence in half and the request is rather esoteric, so feel free to discard this comment. i just needed to get that thought out of my head. :)

Reply:
This is good feedback. This sentence is way too long.